index.php

```diff
@@ -1,26 +1,24 @@
 <?php
 require('vendor/autoload.php');
-define('AMQP_DEBUG', true);
-use PhpAmqpLib\Connection\AMQPConnection;
-use PhpAmqpLib\Message\AMQPMessage;
-$url = parse_url(getenv('CLOUDAMQP_URL'));
-$conn = new AMQPConnection($url['host'], 5672, $url['user'], $url['pass'], substr($url['path'], 1));
-$ch = $conn->channel();
 
-$exchange = 'amq.direct';
-$queue = 'basic_get_queue';
-$ch->queue_declare($queue, false, true, false, false);
-$ch->exchange_declare($exchange, 'direct', true, true, false);
-$ch->queue_bind($queue, $exchange);
+use Enqueue\AmqpBunny\AmqpConnectionFactory;
+use Interop\Amqp\AmqpMessage;
+use Interop\Amqp\AmqpQueue;
 
-$msg_body = 'the body';
-$msg = new AMQPMessage($msg_body, array('content_type' => 'text/plain', 'delivery_mode' => 2));
-$ch->basic_publish($msg, $exchange);
+$context = (new AmqpConnectionFactory(getenv('CLOUDAMQP_URL')))->createContext();
+$queue = $context->createQueue('basic_get_queue');
+$queue->addFlag(AmqpQueue::FLAG_DURABLE);
 
-$retrived_msg = $ch->basic_get($queue);
-var_dump($retrived_msg->body);
-$ch->basic_ack($retrived_msg->delivery_info['delivery_tag']);
+$message = $context->createMessage('the body');
+$message->setContentType('text/plain');
+$message->setDeliveryMode(AmqpMessage::DELIVERY_MODE_PERSISTENT);
 
-$ch->close();
-$conn->close();
-?>
+$context->createProducer()->send($queue, $message);
```

Comment:
what's a producer? a channel? so you create a single channel for each publish?

Reply:
> what's a producer?

https://docs.oracle.com/javaee/7/api/javax/jms/JMSProducer.html

The most important parts from there:

```
A JMSProducer is a simple object used to send messages on behalf of a JMSContext.
```

```
Instances of JMSProducer are intended to be lightweight objects which can be created freely and which do not consume significant resources. This interface therefore does not provide a close method.
```

> a channel?

The producer is not a channel. The context is a representation of AMQP's channel. 

https://docs.oracle.com/javaee/7/api/javax/jms/JMSContext.html